src/datasources/floodscan.py

```diff
@@ -19,7 +23,240 @@
+        df_exp_adm_existing = pd.DataFrame(columns=["date"])
+
+    existing_dates = df_exp_adm_existing["date"].unique()
+    # filter to only unprocessed exposure rasters
+    unprocessed_exposure_rasters = [
+        x
+        for x in recent_exposure_rasters
+        if datetime.strptime(x.split("/")[-1][13:23], "%Y-%m-%d")
+        not in existing_dates
+        or clobber
+    ]
+    if clobber:
+        df_empty = pd.DataFrame(columns=["date"])
+        blob_name = get_blob_name(iso3, "exposure_tabular")
+        blob.upload_parquet_to_blob(blob_name, df_empty, index=False)
+
+    # break list of exposure rasters into chunks, to avoid memory issues
+    # chunk length is arbitrary, but seems to work fine
+    chunk_len = 100
+    exposure_raster_chunks = [
+        unprocessed_exposure_rasters[x : x + chunk_len]
+        for x in range(0, len(unprocessed_exposure_rasters), chunk_len)
+    ]
+
+    # iterate over chunks
+    for exposure_raster_chunk in tqdm(exposure_raster_chunks):
+        blob_name = get_blob_name(iso3, "exposure_tabular")
+        try:
+            df_exp_adm_existing = blob.load_parquet_from_blob(blob_name)
+        except Exception as e:
+            print(e)
+            df_exp_adm_existing = pd.DataFrame(columns=["date"])
+
+        existing_dates = df_exp_adm_existing["date"].unique()
+        if verbose:
+            print(existing_dates)
+
+        # stack up exposure rasters in chunk
+        das = []
+        for blob_name in tqdm(exposure_raster_chunk):
+            date_in = datetime.strptime(
+                blob_name.split("/")[-1][13:23], "%Y-%m-%d"
+            )
+            if date_in in existing_dates and not clobber:
+                if verbose:
+                    print(f"already processed for {date_in}, skipping")
+                continue
+            try:
+                da_in = blob.open_blob_cog(blob_name)
+                da_in["date"] = date_in
+                da_in = da_in.persist()
+                das.append(da_in)
+            except Exception as e:
+                print(e)
+                print(f"couldn't open {blob_name}")
+
+        if len(das) == 0:
+            print("all complete for chunk")
+            continue
+        ds_exp_recent = xr.concat(das, dim="date").squeeze(
+            dim="band", drop=True
+        )
+        if verbose:
+            print(ds_exp_recent)
+
+        # iterate over admin level 2 regions and calculate exposure sums
+        dfs = []
+        for pcode, row in tqdm(
+            adm.set_index("ADM2_PCODE").iterrows(), total=len(adm)
+        ):
+            da_clip = ds_exp_recent.rio.clip([row.geometry])
+            dff = (
+                da_clip.sum(dim=["x", "y"])
+                .to_dataframe(name="total_exposed")["total_exposed"]
+                .astype(int)
+                .reset_index()
+            )
+            dff["ADM2_PCODE"] = pcode
+            dfs.append(dff)
+
+        df_exp_adm_new = pd.concat(dfs, ignore_index=True)
```

Comment:
And here is where we're calculating the raster stats (in this case, just `sum`). But I guess we should replace this with whatever standard method we are using in `ds-raster-stats`

Reply:
Yeah potentially worth considering in the future, but for now I think 1km is ok (all the other raster stats are 0.05deg, so around 5km).